### PR TITLE
Feat/main menu and level select

### LINKED ISSUE #159: Feature: Main Menu Screen
Criteria:
- Must have start game and exit game button
- Start game button links you to a level select screen
- Exit game button quits the game
- Can be ugly grey boxes for now

### LINKED ISSUE #160: Feature: Level Select Screen
Criteria:
- [x] Button for every level
- [ ] some levels are uninteractable because you havent finished the level before
- [x] When selecting a level display best grade and description of level.
- [ ] After selecting a level a button is activated saying either start / resit

### LINKED ISSUE #161: Feature: Grading System
Criteria:
- [x] The end of level screen needs to show the grade you got for your geisting performance
- [x] Ask designer what variables influence the grade and how much
- [x] Keep track of these variables and show them in the end screen for the level
- [ ] Level end screen should include three buttons: "Resit" (Restart), Next assignment (Next level) and Level Select 
- [x] Best grade achieved (with the tracked variables) per level should be saved.

### PR DESCRIPTION
**Pull request by Jeroenvn**
## Description
Created main menu, settings and level select screen.
Created system for writing and reading grades into and from files.
Created ui element for displaying grade scores.

## Setting up testing environment
1. In the project window navigate to Project/Assets/Scenes and open the "Mansion" scene.
2. Press Play.

#### Main menu
Criteria:
- [x] play button opens level select
- [x] settings button opens settings
- [x] quit button prints quit game to console

#### Settings
Criteria:
- [x] contains multiple working sliders
- [x] back button returns to main menu

#### Level select
Criteria:
- [x] displays all levels that are in the LevelCatalog scriptable gameobject
- [x] when selecting a level you see a report / grade card on the right with details of your highest score
- [x] if you've already finished the mansion scene once this report / grade card shows your saved grade

#### Grade system
Criteria:
- [x] After completing or failing a level a report card is displayed with your grades which reflect your gameplay
- [x] These grades acurately display your experience except for the phobia scares
- [x] If you achieve a new highest grade it gets saved to a text file in your project folder at Assets/Scripts/Game/Grades

## Checklist
_These checks should always be true for your pull request_
- [x] My pull request is for one story/feature.
- [x] Every individual commit in this pull request is logical.
- [x] All code, documentation, commits and player seen texts are in English.
- [x] I have deleted unused / unnecessary code.
- [x] If my edits need a change in documentation, then I have updated the documentation.
- [x] All code is in correspondence with the code conventions.

### CODE REVIEW

File: Assets/Scripts/Game/LevelGradeHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+using System;
+using System.IO;
+using UnityEngine;
+
+public class LevelGradeHandler
+{
+    private string _gradeDirPath = "Assets/Scripts/Game/Grades";
```
Comment (ThijsvanDijk) [inline Assets/Scripts/Game/LevelGradeHandler.cs:7]: Will this also work when building?

File: Assets/Scripts/UI/MainMenuController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    private void OnSettingsButtonPressed()
+    {
+        _settingsCanvas.SetActive(true);
+        gameObject.SetActive(false);
+    }
+
+    private void OnExitButtonPressed()
+    {
+        Debug.Log("Exit Game");
```
Comment (ThijsvanDijk) [inline Assets/Scripts/UI/MainMenuController.cs:39]: Can probably be removed

**Review by ThijsvanDijk — APPROVED** [review]
It looks good and works well, but we probably need to change the location of where it stores your grades when we want to do an actual build

File: Assets/Scripts/Game/LevelGradeHandler.cs (hunk 1 above)
Comment (Jeroenvn) [reply Assets/Scripts/Game/LevelGradeHandler.cs:7]: good question, I'll look into it

Comment (Jeroenvn) [reply Assets/Scripts/Game/LevelGradeHandler.cs:7]: image
So um, it works. It just creates the folder where it should put it.

**Pull request merged**